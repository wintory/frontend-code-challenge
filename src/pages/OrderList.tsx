import { FC, useEffect, useState } from 'react'

import OrderBook from '../containers/OrderBook/OrderBook'
import OrderForm from '../containers/OrderForm/OrderForm'
import OrderHeader from '../containers/OrderHeader/OrderHeader'

import useOrderBookStream from '../hooks/useOrderBookStream'

import { TOrderBookStream } from '../types/orderBook'
import { TOrderForm } from '../types/orderForm'

const OrderList: FC = () => {
  const [orderBook, setOrderBook] = useState<TOrderBookStream>({
    buy: [],
    sell: [],
  })
  const { subscribe, addOrder } = useOrderBookStream()

  useEffect(() => {
    subscribe((data: TOrderBookStream) => {
      setOrderBook(data)
    })

    // Todo: setup lint to ignore react-hooks/exhaustive-deps
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])

  const handleSubmitOrder: TOrderForm['submitOrder'] = (side, { price, amount }) => {
    addOrder(side, { price, amount })
  }

  return (
    <>
      <OrderHeader />
      <OrderBook buy={orderBook.buy} sell={orderBook.sell} />
      <OrderForm submitOrder={handleSubmitOrder} />
    </>
  )
}

export default OrderList
