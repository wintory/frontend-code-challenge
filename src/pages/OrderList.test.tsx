import { render } from '@testing-library/react'

import OrderList from './OrderList'

describe('OrderList', () => {
  it('should render if exists', () => {
    const { getAllByText } = render(<OrderList />)

    expect(getAllByText('Price')).toBeTruthy()
  })

  it('should match snapshot', () => {
    const { container } = render(<OrderList />)

    expect(container.firstChild).toMatchSnapshot()
  })

  // Todo: add more unit test
})
