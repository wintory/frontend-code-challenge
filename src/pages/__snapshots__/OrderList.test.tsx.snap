// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderList should match snapshot 1`] = `
<div
  class="MuiGrid-root sc-jRQBWg eUNOvD css-18qtskw-MuiGrid-root"
>
  <div
    class="MuiGrid-root MuiGrid-item css-y526ko-MuiGrid-root"
  >
    <img
      alt="logo"
      class="sc-eCImPb jYisjZ App-logo"
      src="/assets/logo/zipmex-logo.png"
    />
  </div>
</div>
`;
